challenge-13/challenge-13.js

```diff
@@ -1,150 +1,226 @@
+    */
+    console.log( '\nEstados do Sul do Brasil:' );
+
+    console.log( newSul );
+
+    /*
+    Mostre no console todos os estados que estão em `brasil`.
+    */
+    console.log( '\nAlguns Estados do Brasil:' );
+    console.log( brasil );
+
+    /*
+    Crie um novo array chamado `nordeste`, que tenha os estados do nordeste.
+    */
+    var nordeste =[
+          'Maranhão',
+          'Piaui',
+          'Bahia',
+          'Alagoas',
+          'Ceará',
+          'Paraiba',
+          'Rio Grande do Norte',
+          'Pernambuco'
+     ];
+
+    /*
+    Mostre no console os estados do nordeste.
+    */
+    console.log( '\nEstados do Nordeste:' );
+
+    console.log( nordeste );
+
+    /*
+    Remova de `brasil` os estados do `sudeste`, colocando-os em uma variável
+    chamada `newSudeste`.
+    */
+    var newSudeste = brasil.splice( 2, 4 + 1 );
+
+    /*
+    Adicione os estados do `nordeste` ao array `brasil`. Esses estados devem
+    ficar no mesmo nível que os estados já existentes, não em um array separado.
+    */
+    brasil = brasil.concat( nordeste );
+
+    /*
+    Mostre no console os estados em `newSudeste`.
+    */
+    console.log( '\nEstados em newSudeste:' );
+
+    console.log( newSudeste );
+
+    /*
+    Mostre no console os estados do `brasil`.
+    */
+    console.log( '\nAlguns estados do Brasil:' );
+    console.log( brasil );
+
+    /*
+    usando forEach, percorra o array `brasil` e gere um novo array chamado
+    `newBrasil`. Esse array deve ter cada item como um objeto, com as
+    propriedades:
+    - `id`: que será o índice do array `brasil`,
+    - `estado`: que será o estado do array `brasil`.
+    */
+
+    var newBrasil = [];
+    brasil.forEach(function(  item, index ){
+        newBrasil.push({ 'id': index, 'estado': item});
+    });
+
+    /*
+    Mostre o array `newBrasil` no console
+    */
+    console.log( '\nnewBrasil:' );
+    console.log( newBrasil );
+
+    /*
+    Percorra o array `brasil` e verifique se os estados tem mais de 7 letras cada,
+    atribuindo o resultado à uma variável. Se tiver, mostre no console a frase:
+    - "Sim, todos os estados tem mais de 7 letras!"
+    Senão, mostre no console:
+    - "Nem todos os estados tem mais de 7 letras!"
+    */
+    console.log( '\nTodos os estados de `brasil` tem mais de 7 letras?' );
+
+    var sevenLetters = brasil.every(function( item ){
+      return item.length > 7;
+    });
+    console.log( sevenLetters ? 'Sim, todos os estados tem mais de 7 letras!' :
+      'Nem todos os estados tem mais de 7 letras!');
+
+    /*
+    fiz essa outra forma,acho que fica mais legivel para quem
+    não conhece o metodo every().
+    */
+
+   var init = 0;
+   var sevenLetters = 0;
+
+   for(var i = 0; i <= brasil.length; i++) {
+      init = brasil.length;
+      sevenLetters += brasil.length;
+   }
+    console.log( init * 7 > sevenLetters  ? 'Sim, todos os estados tem mais de 7 letras!' :
+      'Nem todos os estados tem mais de 7 letras!');
```

Comment:
Boa! É mais questão gosto, mas com o `every` fica mais simples e explícito o que você está testando. Ainda que alguém não o conheça, fica mais fácil pesquisar só para saber o que realmente ele faz, mas o próprio método deixa muito claro por causa do nome dele :)

Mas como você fez é uma boa abordagem, dá pra usar também sem problemas :)

Só o `init` que não precisaria atribuir a cada iteração, já que o valor de `brasil.length` é um valor fixo, nunca vai mudar. Poderia ter atribuído fora do `for`, no lugar do zero :)